tests/utils/em.py

```diff
@@ -0,0 +1,313 @@
+import itertools
+
+# The utilization rate limit to switch to the overutilization mode
+OVERUTILIZED_RATE = 0.8
+
+def _get_powers_list(node):
+    """
+    Get the list of powers for a given node.
+    :param node: A node of the energy model. It can be a cluster or cpu.
+    :type node: EnergyModelNode
+
+    :returns: The list of powers associates with the given node.
+    """
+    powers = []
+    for freq in node.active_states.keys():
+        powers.append(node.active_states[freq].power)
+    return powers
+
+def _get_capacities_list(node):
+    """
+    Get the list of capacities for a given node.
+    :param node: A node of the energy model. It can be a cluster or cpu.
+    :type node: EnergyModelNode
+
+    :returns: The list of capacities associates with the given node.
+    """
+    cap = []
+    for freq in node.active_states.keys():
+        cap.append(node.active_states[freq].capacity)
+    return cap
+
+def _is_list_strickly_increasing(l):
+    return all(x<y for x, y in zip(l, l[1:]))
```

Comment:
Done --> I changed it for a for loop

Reply:
all() solution is slowed down by having to call next() on the iterator for every new item (and restoring its state). Using a list comprehension solves that.
The for loop is fast because it does not need to have this iterator state to be restored every time and is similar to all([...]).
The fastest solution of all (on a sorted list) is a plain "l == sorted(l)". This is because it just involves 2 calls into C code (sorted builtin and comparison), instead of executing Python. CPython is so slow that anything that reduces the number of opcodes executed will reduce the runtime, regardless of the algorithmic complexity of what is actually done:
              0 LOAD_FAST                   0 (l)
              2 LOAD_GLOBAL             0 (sorted)
              4 LOAD_FAST                   0 (l)
              6 CALL_FUNCTION         1
              8 COMPARE_OP              2 (==)
             10 RETURN_VALUE